### PR TITLE
Part 2: Generate webpack manifest and add asset helper

### PR DESCRIPTION
**Pull request by davidsbailey**
### Background

See [webpack content hashing design doc](https://docs.google.com/document/d/1vJ3EpaT8ch6UMLP21IovZIgOzS-QhU4G1kv3azZPmZo/edit#)

### Description

This PR contains the following steps:
* let webpack start generating a manifest.json file 
* create a `webpack_asset_path` helper which can be used by both dashboard and pegasus

### CODE REVIEW

File: lib/test/fixtures/webpack-manifest.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "js/webpack-runtime.js": "/blockly/js/webpack-runtime.338c9dacc207fe9b1eb4.min.js",
+  "js/vendors.js": "/blockly/js/vendors.b54aa8aa165e50793d97.min.js",
```
Comment (davidsbailey) [inline lib/test/fixtures/webpack-manifest.json:3]: this manifest fixture is borrowed from a future PR, which is why it contains content hashes. the code in this PR would generate a manifest which looks more like this:
```
{
  "js/webpack-runtime.js": "/blockly/js/webpack-runtime.min.js",
  "js/vendors.js": "/blockly/js/vendors.min.js",
```

File: apps/webpack.js [hunk 2]
```diff
@@ -321,16 +321,17 @@ function create(options) {
   var optimization = options.optimization;
   var mode = options.mode;
 
+  // When debugging minified code, use the .js suffix (rather than .min.js)
+  // to allow the application to load minified js locally without running it
+  // through the rails asset pipeline. This is much simpler than hacking the
+  // application to load .min.js locally.
```
Comment (islemaster) [inline apps/webpack.js:327]: This comment moved; is it still accurate? Specifically wondering if the "without running it through the rails asset pipeline" comment still makes sense.

**Review by islemaster — APPROVED** [review]
This is an interesting stopping point, given we're adding some unused code and unused generated assets, but it does help with reviewing.  Thanks!

File: apps/webpack.js (hunk 2 above)
Comment (davidsbailey) [reply apps/webpack.js:327]: great catch! it is no longer necessary to run the rails asset pipeline in order to use minified, digested js locally. I will update these comments.

**Comment by davidsbailey** [comment]
> This is an interesting stopping point, given we're adding some unused code and unused generated assets, but it does help with reviewing. Thanks!

The stopping point is a bit contrived, but my goal here was to have as little as possible to revert if something goes wrong in the next PR (which is still quite a lot).

File: apps/webpack.js (hunk 2 above)
Comment (davidsbailey) [reply apps/webpack.js:327]: Thanks again Brad, I was able to remove some functionality related to debugMinify. I also made some major updates to the instructions and documentation in build.md. You can see them on this commit which I have added to the next PR in the sequence: 28b7d5ca397

Comment (davidsbailey) [reply apps/webpack.js:327]: Hi Brad, apologies for the rebase. Bumping this comment thread to see if the updates in 28b7d5c seem sufficient.

Comment (islemaster) [reply apps/webpack.js:327]: Yes! Awesome update to our documentation.  Reading through that clarified a lot of things in my mental model of what we're changing.  Thank you!

**Pull request merged**